code/datums/status_effects/buffs.dm

```diff
@@ -151,3 +151,49 @@
 		return
 	playsound(cyborg, 'sound/effects/light_flicker.ogg', 50, 1)
 	cyborg.cell.give(power_to_give)
+
+/datum/status_effect/his_grace
+	id = "his_grace"
+	duration = -1
+	tick_interval = 5
+	alert_type = /obj/screen/alert/status_effect/his_grace
+	var/bloodlust = 0
+
+/obj/screen/alert/status_effect/his_grace
+	name = "His Grace"
+	desc = "His Grace hungers, and you must feed Him."
+	icon_state = "his_grace"
+	alerttooltipstyle = "hisgrace"
+
+/obj/screen/alert/status_effect/his_grace/MouseEntered(location,control,params)
+	desc = initial(desc)
+	var/datum/status_effect/his_grace/HG = attached_effect
+	desc += "<br><font size=3><b>Current Bloodthirst: [HG.bloodlust]</b></font>\
+	<br>Becomes undroppable at <b>[HIS_GRACE_FAMISHED]</b>\
+	<br>Will consume you at <b>[HIS_GRACE_CONSUME_OWNER]</b>"
+	..()
+
+/datum/status_effect/his_grace/on_apply()
+	add_logs(owner, null, "gained His Grace's stun immunity")
+	owner.add_stun_absorption("hisgrace", INFINITY, 3, null, "His Grace protects you from the stun!")
+
+/datum/status_effect/his_grace/tick()
+	bloodlust = 0
+	for(var/obj/item/weapon/his_grace/HG in owner.held_items)
+		if(HG.bloodthirst > bloodlust)
+			bloodlust = HG.bloodthirst
+	if(!bloodlust)
+		qdel(src)
+		return
+	var/grace_heal = Floor(bloodlust * 0.08)
+	owner.adjustBruteLoss(-grace_heal)
+	owner.adjustFireLoss(-grace_heal)
+	owner.adjustToxLoss(-grace_heal)
```

Comment:
Owner could be TOXINLOVING. We really need some sort of standard proc or flag for ignoring this sort of stuff. Tbh, I've never been a fan of TOXINLOVING, it makes no sense.